eng/pipelines/templates/jobs/ci.tests.yml

```diff
@@ -68,6 +68,18 @@ jobs:
       value: ""
     - name: AZURE_TEST_MODE
       value: "PLAYBACK"
+# Even on playblack, tests will require ENV VARS
```

Comment:
cc @benbp 

Do we need to specify these explicitly? Shouldn't these be set in the tests or default correctly without needing to be set by the pipeline?

Reply:
The way I see it, a pipeline is representing an automated user, hence, since it is user's responsibility to set this required env, the pipeline should be setting this up.

For the nightly pipeline, the pipeline sets the TEST_MODE to LIVE and set the required env vars for tests to talk to the deployed test resources.

For CI/public pipeline, the pipeline knows about the recordings and sets TEST_MODE=PLAYBACK.  But it should also provide the required env vars.  ( test cases should never know if they are running on PLAYBACK or not, that's something only the person running the tests should know and set up )

Reply:
From my prospective for local running of these tests it would be ideal if folks didn't need to know all these env variables to set in order to get playback tests to work. They should just be able to run the tests in playback mode and things should just work, if they have to set a bunch of other things to make them work it raises the bar for getting starting running tests in our repo. I will say for other repos like .NET that isn't the case, playback mode is the default and so anyone just building and running the tests will be able to do so without any extra setup.

Reply:
that's a valid point. 
We used to have default values to facilitate running on PLAYBACK.  However, PLAYBACK mode is not the default.
So, running tests won't throw with something like `Please set up a SERVICE_URL env var for the test`, but instead, we would throw with something like `Failed to connect to testUrlForRecording`.

We could switch to make PLAYBACK the default. It would make running tests cases easier the first time ( no need to lean about setting up record-playback), but it would just move the complexity to the first-time people adding a test case ( test will fail because the recording is not found, and now people need to learn how to set recording ON. )

I will follow up with the team anyway, for how we want the defaults to be. Then we can re-visit this pipeline configs and make changes.

Reply:
As a separate point, specifying service specific values in common config places has been a config bloat headache for us in the other repositories that is taking a long time to unravel. For values like `AZURE_KEYVAULT_URL` if they need to be defaulted, they should go in the respective `ci.yml` files (which makes it more obvious to pipeline users where the relevant values are set anyway).

Re: running tests in live mode locally, the happy path is to use the `New-TestResources.ps1` script and bicep templates, and either setting the `AZURE_*` values in the bicep template or looking for both `AZURE_*` and `SERVICE_DIRECTORY_*` formatted values from the tests. These should set the appropriate default values rather than the pipeline yaml, that way tests have better parity between pipeline and local execution.

Reply:
you're right @benbp .  we use the `test-resources.json` to define the services and the env vars to be set.

We need to use the ci.yml on each service to set whatever required ENV VARs  . Agree with that.  I will create an issue to make it like that!  :)

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/3161